nomos-blend/network/src/core/with_core/behaviour/mod.rs

```diff
@@ -677,11 +678,11 @@ impl<ProofsVerifier, ObservationWindowClockProvider>
     ) -> Option<NegotiatedPeerState> {
         let peer_details = self.negotiated_peers.get_mut(&peer_id)?;
         // We double check we are dealing with the expected connection.
-        debug_assert!(
-            peer_details.connection_id == connection_id,
-            "Stored connection ID and provided connection ID do not match for the provided peer ID."
-        );
-
+        // This could be false if `connection_id` is from the old session.
+        if peer_details.connection_id != connection_id {
```

Comment:
Don't we want to update the state for a peer in the old connection as well?

Reply:
Since we always try with both current and old session, I think we could do that here as well.

Reply:
This is actually what we need to discuss (maybe with Marcin). In this PR, we ignore monitoring events from the old session. One big reason is that, even if we detects a malicious connection and closes it, we cannot create another connection in our current codebase, because our NetworkBehaviour implementation rejects incoming connection establishments if the dialer is not a core node in the "current" session. In other words, we currently don't support establishing new connections for the old session.
So, I think it's not meaningful to monitor connections in the old session. It might be meaningful to close malicious connections, but I wanted to address this topic as a separate discussion and as a separate PR. Before making any decision, I didn't want to add any temporary logic in this PR.

Reply:
Also, the other parts of the existing codebase also doesn't perform monitoring for the old session. For example, we monitor spammy connections for the current session in the following code:
https://github.com/logos-co/nomos/blob/30eac958fe84c612053a722ef38e2bcb58c4c688/nomos-blend/network/src/core/with_core/behaviour/mod.rs#L847
But we don't do it for the old session:
https://github.com/logos-co/nomos/blob/blend-release-old-msgs-to-old-conns/nomos-blend/network/src/core/with_core/behaviour/old_session.rs#L227

If we want to do monitoring for the old session as well, we can implement it as a separate PR. I think it's fine to not monitor connections for the old session for now because the connections will be closed after STP anyway.